atom/browser/native_window_views.cc

```diff
@@ -486,19 +486,6 @@ void NativeWindowViews::SetFullScreen(bool fullscreen) {
     NotifyWindowLeaveFullScreen();
   }
 
-  // For window without WS_THICKFRAME style, we can not call SetFullscreen().
-  if (!thick_frame_) {
```

Comment:
I think this might still be needed (or some other fix).

On this branch, with the following app, the window becomes unusable once full screen state is left:

```js
const {app, BrowserWindow} = require('electron')

let browserWindow

app.once('ready', function () {
  browserWindow = new BrowserWindow({
    show: true,
    fullscreen: true,
    transparent: true,
    thickFrame: false
  });
  browserWindow.loadURL(`about:blank`)
})
```

Then open the dev tools and run:

```js
require('electron').remote.getCurrentWindow().setFullScreen(false)
```

The window does resize but if you click on it, it isn't usable and will disappear from the taskbar.